### PR TITLE
DataForm: add unit tests

### PR DESCRIPTION
**Pull request by gigitux**
## PR Description
This PR introduces unit tests for the DataForm component, covering the following use cases:

- Panel mode
- Regular mode
- Combined fields
- Custom Edit component
- Custom render component
- onChange callback testing

Also, this PR improves the `tsconfig.json` to ensure that js test files are skipped.

## Testing Instructions
Ensure that unit test job is green.

## Screenshots or screencast 

|Before|After|
|-|-|
|||

### CODE REVIEW

File: packages/dataviews/tsconfig.json
Comment (gigitux) [inline packages/dataviews/tsconfig.json:1]: I copied pasted the changes from https://github.com/WordPress/gutenberg/blob/bc4b839cbca7c6bc5b5e41dc2859b36599267400/packages/components/tsconfig.json#L1

File: packages/dataviews/src/test/dataform.tsx [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+			} ),
+	},
+};
+
+describe( 'DataForm component', () => {
+	describe( 'in regular mode', () => {
+		it( 'should display fields', () => {
+			render(
+				<Dataform
+					onChange={ () => void 0 }
```
Comment (oandregal) [inline packages/dataviews/src/test/dataform.tsx]: Why not just using `() => {}` as a noop? The `void` operator is not very common in our codebase.

Comment (gigitux) [reply packages/dataviews/src/test/dataform.tsx]: It makes sense. Fixed with [238f654](https://github.com/WordPress/gutenberg/pull/68054/commits/238f654931b5e50ed1bc1f21ca1557964d4927a7).

File: packages/dataviews/src/test/dataform.tsx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+					data={ data }
+				/>
+			);
+
+			expect(
+				// It is used here to ensure that the fields are wrapped in HStack. This happens when the labelPosition is set to side.
+				// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+				container.querySelectorAll( "[data-wp-component='HStack']" )
+			).toHaveLength( 3 );
+		} );
```
Comment (oandregal) [inline packages/dataviews/src/test/dataform.tsx]: This kind test is too implementation-specific in my view. We're testing a specific HTML output that we may need to change. Personally, I find tests more useful when they test the API. Don't feel you need to remove it, but I find it a bit fragile.

**Review by oandregal — APPROVED** [review]
Thanks. It'd be great to have some e2e tests for the QuickEdit experiment as well :)

File: packages/dataviews/src/test/dataform.tsx (hunk 2 above)
Comment (gigitux) [reply packages/dataviews/src/test/dataform.tsx]: 💯 agree with you, but I didn't find a better way to test this behavior.

**Comment by gigitux** [comment]
> Thanks. It'd be great to have some e2e tests for the QuickEdit experiment as well :)

Thanks for the review! I'm going to work on it! Thanks for the feedback!

**Pull request merged**